### DISCUSSION TITLE
[FEATURE REQUEST] salt.states.rsync.synchronized

### DISCUSSION THREAD

**Discussion by jpisard** [saltstack/salt#67501] [Feature Requests]
use salt.states.rsync.synchronized with rsync server (server::sharedfolder ) as destination could be a probleme.

we must use "force" parameter else the state don't find destination directory.
In these case the "prepare" option create a local directory.

I would like a parameter to do rsync without the rsync --force option

i have make my state with : 

```
def synchronized(name, source,

                 delete=False,
                 force=False,
                 update=False,
                 passwordfile=None,
                 exclude=None,
                 excludefrom=None,
                 testfolder=True,
                 prepare=False,
                 dryrun=False,
                 additional_opts=None):
    '''
    Guarantees that the source directory is always copied to the target.

    name
        Name of the target directory.

    source
        Source directory.

    prepare
        Create destination directory if it does not exists.

    delete
        Delete extraneous files from the destination dirs (True or False)

    force
        Force deletion of dirs even if not empty

    update
        Skip files that are newer on the receiver (True or False)

    passwordfile
        Read daemon-access password from the file (path)

    exclude
        Exclude files, that matches pattern.

    excludefrom
        Read exclude patterns from the file (path)

    dryrun
        Perform a trial run with no changes made. Is the same as
        doing test=True

        .. versionadded:: 2016.3.1

    additional_opts
        Pass additional options to rsync, should be included as a list.

        .. versionadded:: 2018.3.0
    '''

    ret = {'name': name, 'changes': {}, 'result': True, 'comment': ''}

    if not os.path.exists(name) and not force and not prepare and testfolder:

```

**Comment by welcome** [comment]
Hi there! Welcome to the Salt Community! Thank you for making your first contribution. We have a lengthy process for issues and PRs. Someone from the Core Team will follow up as soon as possible. In the meantime, here’s some information that may help as you continue your Salt journey.
Please be sure to review our [Code of Conduct](https://github.com/saltstack/salt/blob/master/CODE_OF_CONDUCT.md). Also, check out some of our community resources including:

  - [Community Wiki](https://github.com/saltstack/community/wiki)
  - [Salt’s Contributor Guide](https://docs.saltstack.com/en/latest/topics/development/contributing.html)
  - [Join our Community Slack](https://saltstackcommunity.herokuapp.com/)
  - [IRC on Freenode](https://webchat.freenode.net/#salt)
  - [SaltStack YouTube channel](https://www.youtube.com/user/SaltStack)
  - [SaltStackInc Twitch channel](https://www.twitch.tv/saltstackinc)

There are lots of ways to get involved in our community. Every month, there are around a dozen opportunities to meet with other contributors and the Salt Core team and collaborate in real time. The best way to keep track is by subscribing to the Salt Community Events Calendar.
If you have additional questions, email us at core@saltstack.com. We’re glad you’ve joined our community and look forward to doing awesome things with you!

**Comment by s0undt3ch** [comment]
Since you seem to already have a solution, care to submit a PR?
